src/Orleans/Streams/Internal/StreamSubscriptionHandleImpl.cs

```diff
@@ -79,14 +86,16 @@ public override Task<StreamSubscriptionHandle<T>> ResumeAsync(IAsyncObserver<T>
                 await NextItem(itemTuple.Item1, itemTuple.Item2);
             }
 
-            expectedToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken);
-
+            if (isRewindable)
+            {
+                expectedToken = StreamHandshakeToken.CreateDeliveyToken(batch.SequenceToken);
+            }
             return null;
         }
 
         public async Task<StreamHandshakeToken> DeliverItem(object item, StreamSequenceToken currentToken, StreamHandshakeToken handshakeToken)
         {
-            if (expectedToken != null)
+            if (isRewindable && expectedToken != null)
```

Comment:
This line is the crux of the fix.
Initial, original implementation of AQST did not have this check. It was added later on for EH but was not conditioned on the stream provider being ordered/rewindable.